File: 백준/Gold/13164. 행복 유치원/행복 유치원.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *

setrecursionlimit(10**9)
input = stdin.readline


n,k = map(int,input().split())

arr = list(map(int,input().split()))


heightGap = []

for i in range(1,n):
    heightGap.append(arr[i] - arr[i-1])
    

heightGap.sort(reverse=True)
print(sum(heightGap[k-1:]))
        

    
    

File: 백준/Gold/13164. 행복 유치원/행복 유치원.swift
let nk = readLine()!.split{$0 == " "}.map{Int($0)!}
let (n,k) = (nk[0],nk[1])

let arr = readLine()!.split{$0 == " "}.map{Int($0)!}

var result: Int = 0

var gap: [Int] = []

for i in 1..<n {
    gap.append(arr[i]-arr[i-1])
}

gap.sort(by:>)

print(Array(gap[k-1..<gap.count]).reduce(0,+))

